addons/html_editor/static/src/core/selection_plugin.js

```diff
@@ -776,12 +785,96 @@ export class SelectionPlugin extends Plugin {
+                    domDirection,
+                    this.editable
+                );
 
                 shouldSkip =
                     hasSelectionChanged &&
                     shouldSkipCallbacks.some((cb) => cb(ev, adjacentCharacter, lastSkippedChar));
             }
+        } else if (["ArrowUp", "ArrowDown"].includes(ev.key)) {
+            const { anchorNode, anchorOffset } = this.getSelectionData().deepEditableSelection;
+            const block = closestBlock(anchorNode);
+            const isArrowUp = ev.key === "ArrowUp";
+            const currentNode = anchorNode === block ? block.childNodes[anchorOffset] : anchorNode;
+
+            // Check if all child or sibling nodes are icons or non-visible text
+            const areAllIcons = (element) => {
+                if (!element) {
+                    return false;
+                }
+                let node =
+                    isBlock(element) && element.childNodes
+                        ? isArrowUp
+                            ? element.lastChild
+                            : element.firstChild
+                        : isArrowUp
+                        ? element.previousSibling
+                        : element.nextSibling;
+                let foundIcon = false;
+                while (node) {
+                    if (isIconElement(node) || (isTextNode(node) && !isVisibleTextNode(node))) {
+                        foundIcon = true;
+                    } else if (
+                        foundIcon &&
+                        node.nodeType === Node.ELEMENT_NODE &&
+                        node.tagName === "BR"
+                    ) {
+                        return true;
+                    } else {
+                        return false;
+                    }
+                    node = isArrowUp ? node.previousSibling : node.nextSibling;
+                }
+                return foundIcon;
+            };
+
+            // Find the nearest <br> element
+            const findLineBreak = (element) => {
+                let node = element || currentNode;
+                while (node) {
+                    if (node.nodeType === Node.ELEMENT_NODE && node.tagName === "BR") {
+                        return node;
+                    }
+                    node = isArrowUp ? node.previousSibling : node.nextSibling;
+                }
+                return null;
+            };
+
+            const lineBreakNode = findLineBreak();
+            if (isArrowUp) {
+                const previousLineBreak = findLineBreak(
+                    isIconElement(currentNode) ? currentNode : lineBreakNode?.previousSibling
+                );
+                if (
+                    (lineBreakNode && previousLineBreak && areAllIcons(previousLineBreak)) ||
+                    ((currentNode.tagName === "BR" ||
+                        (isIconElement(currentNode) && previousLineBreak)) &&
+                        areAllIcons(currentNode))
+                ) {
```

Comment:
I think this can be written as:
```suggestion
                if (lineBreakNode && isIconElement(currentNode)) {
```
I checked and the bug only occurs when `currentNode` is iconElement when pressing down key.